backend/amd64/reload.ml

```diff
@@ -108,9 +108,9 @@ method! reload_operation op arg res =
         if stackp res.(0) then [| self#makereg res.(0) |] else res
       in
       arg, res
-  | Iintop(Imulh _ | Idiv | Imod | Ilsl | Ilsr | Iasr)
+  | Iintop(Imulh _ | Idiv | Imod | Ilsl | Ilsr | Iasr | Icheckalign _)
```

Comment:
Why is Icheckalign here, but Icheckbound was not in the existing code?

Reply:
Unlike `Icheckbound`, `Icheckalign` always has an immediate second argument, so it doesn't need to make sure that only one arg is spilled. I don't think we assert that checkalign is actually immediate, though, so I think this could break if someone used checkalign with a temp - moved it to the same case as `Icheckboudn` since it doesn't hurt.